### ISSUE TITLE
useless use of `format!`: should return function directly

### ISSUE THREAD

**Issue by Dentrax** [rust-lang/rust-clippy#7066]
I tried this code:

```bash
$ cargo clippy --all-targets --all-features -- -D clippy::all
```

1. Linter says:
```rust
let v: Vec<String> = vec!["foo".to_string(), "bar".to_string()];
let s: String = format!("{}", &*v.join("\n"));
// useless use of `format!`: help: consider using `.to_string()`: `&*v.join("\n").to_string()`
```

2. After fix what linter says:
```rust
let v: Vec<String> = vec!["foo".to_string(), "bar".to_string()];
let s: String = &*v.join("\n").to_string();
// IDE error: mismatched types [E0308] expected `String`, found `&<unknown>`
// Linter error: mismatched types: expected struct `std::string::String`, found `&str`
```

3. Manually fix:
```rust
v.join("\n").to_string()
//           ^^^^^^^^^^^^ redundant clone: help: remove this
```

I expected to see this happen: *explanation*

```
help: consider using `v.join("\n")`
```

Instead, this happened: *explanation*

```
useless use of `format!`: help: consider using `.to_string()`: `&*v.join("\n").to_string()`
```

### Meta

- `cargo clippy -V`: `clippy 0.1.51 (2fd73fab 2021-03-23)`
- `rustc -Vv`:
```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-apple-darwin
release: 1.51.0
LLVM version: 11.0.1
```

**EDIT:** _Accidiantelly opened issue format as `false-negative`, can someone please change label to `false-positive`?_

**Comment by giraffate** [comment]
In this case, I think it would be better to suggest `(&*v.join("\n")).to_string()`, not `&*v.join("\n").to_string()`.
